src/engine/rendering/Shader.cpp

```diff
@@ -111,37 +111,31 @@ namespace lei3d
 
 	void Shader::setUniformMat4(const std::string& name, const glm::mat4& matrix) const
 	{
-		bind();
```

Comment:
@kevinsadi @Bowers-L 
I removed calls to use shader program when setting uniforms, to reduce calls to the OpenGL state for overhead and to help me sift through commands when GPU debugging.
Made all relevant fixes. Pls don't put back.